[CI] TransportServiceHandshakeTests testAcceptsMismatchedServerlessBuildHash failing

**Build Scans:**
- [elasticsearch-periodic #6363 / openjdk24_entitlements_false_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/3x67fjphaqj7e)
- [elasticsearch-periodic #6363 / openjdk24_entitlements_true_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/dkmat7ic6bdeg)
- [elasticsearch-periodic #6356 / openjdk24_entitlements_false_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/wdtlmqehrmlcc)
- [elasticsearch-periodic #6356 / openjdk24_entitlements_true_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/c2oxofyeec2d6)
- [elasticsearch-periodic #6349 / openjdk24_entitlements_false_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/xul5abn7kjs5u)
- [elasticsearch-periodic #6342 / openjdk24_entitlements_false_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/rxkejmm4ahrik)
- [elasticsearch-periodic #6335 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/u7jdlle3fkqv4)
- [elasticsearch-periodic #6328 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/cuf4qx4k7enc2)
- [elasticsearch-periodic #6321 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/ugdu5yhetykuu)
- [elasticsearch-periodic #6314 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/onqxomtzadmuu)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.transport.TransportServiceHandshakeTests.testAcceptsMismatchedServerlessBuildHash" -Dtests.seed=32B2A08BD592F0A8 -Dtests.jvm.argline="-Des.entitlements.enabled=false" -Dtests.locale=ast-ES -Dtests.timezone=BET -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.transport.TransportServiceHandshakeTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testAcceptsMismatchedServerlessBuildHash),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: safeGet: listener was completed exceptionally
```

**Issue Reasons:**
- [main] 18 consecutive failures in test testAcceptsMismatchedServerlessBuildHash
- [main] 4 consecutive failures in step openjdk24_entitlements_false_checkpart1_java-matrix
- [main] 2 consecutive failures in step openjdk24_entitlements_true_checkpart1_java-matrix
- [main] 12 consecutive failures in step openjdk24_checkpart1_java-matrix
- [main] 16 consecutive failures in pipeline elasticsearch-periodic
- [main] 18 failures in test testAcceptsMismatchedServerlessBuildHash (100.0% fail rate in 18 executions)
- [main] 4 failures in step openjdk24_entitlements_false_checkpart1_java-matrix (100.0% fail rate in 4 executions)
- [main] 2 failures in step openjdk24_entitlements_true_checkpart1_java-matrix (100.0% fail rate in 2 executions)
- [main] 12 failures in step openjdk24_checkpart1_java-matrix (100.0% fail rate in 12 executions)
- [main] 16 failures in pipeline elasticsearch-periodic (100.0% fail rate in 16 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

Can reproduce on main

```
  2> REPRODUCE WITH: ./gradlew ":server:test" --tests "org.elasticsearch.transport.TransportServiceHandshakeTests.testAcceptsMismatchedServerlessBuildHash" -Dtests.seed=8278A46E39005C96 -Dtests.jvm.argline="-Des.entitlements.enabled=" -Dtests.locale=en-001 -Dtests.timezone=US/Pacific -Druntime.java=24
  2> java.lang.AssertionError: safeGet: listener was completed exceptionally
        at __randomizedtesting.SeedInfo.seed([8278A46E39005C96:3D3425B1D807E06A]:0)
        at org.elasticsearch.test.ESTestCase.safeGet(ESTestCase.java:2473)
        at org.elasticsearch.test.ESTestCase.safeAwait(ESTestCase.java:2437)
        at org.elasticsearch.test.ESTestCase.safeAwait(ESTestCase.java:2448)
        at org.elasticsearch.transport.AbstractSimpleTransportTestCase.connectToNode(AbstractSimpleTransportTestCase.java:3499)
        at org.elasticsearch.transport.TransportServiceHandshakeTests.testAcceptsMismatchedServerlessBuildHash(TransportServiceHandshakeTests.java:404)

        Caused by:
        java.util.concurrent.ExecutionException: org.elasticsearch.transport.TransportSerializationException: Failed to deserialize response from handler [ContextRestoreResponseHandler[ActionListenerResponseHandler[org.elasticsearch.action.ActionListenerImplementations$DelegatingFailureActionListener/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener/WrappedActionListener{org.elasticsearch.transport.ClusterConnectionManager$$Lambda/0x000007e000496238@3770963}{org.elasticsearch.transport.ClusterConnectionManager$$Lambda/0x000007e000496480@508a6b27}/org.elasticsearch.transport.ClusterConnectionManager$$Lambda/0x000007e0004966c8@18071675/org.elasticsearch.transport.TransportService$$Lambda/0x000007e000497790@6c2a70cc/org.elasticsearch.transport.TransportService$$Lambda/0x000007e000497c50@572d403b]]]
            at org.elasticsearch.action.support.PlainActionFuture$Sync.getValue(PlainActionFuture.java:279)
            at org.elasticsearch.action.support.PlainActionFuture$Sync.get(PlainActionFuture.java:253)
            at org.elasticsearch.action.support.PlainActionFuture.get(PlainActionFuture.java:74)
            at org.elasticsearch.test.ESTestCase.safeGet(ESTestCase.java:2468)
            ... 4 more

            Caused by:
            org.elasticsearch.transport.TransportSerializationException: Failed to deserialize response from handler [ContextRestoreResponseHandler[ActionListenerResponseHandler[org.elasticsearch.action.ActionListenerImplementations$DelegatingFailureActionListener/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener/WrappedActionListener{org.elasticsearch.transport.ClusterConnectionManager$$Lambda/0x000007e000496238@3770963}{org.elasticsearch.transport.ClusterConnectionManager$$Lambda/0x000007e000496480@508a6b27}/org.elasticsearch.transport.ClusterConnectionManager$$Lambda/0x000007e0004966c8@18071675/org.elasticsearch.transport.TransportService$$Lambda/0x000007e000497790@6c2a70cc/org.elasticsearch.transport.TransportService$$Lambda/0x000007e000497c50@572d403b]]]
                at app//org.elasticsearch.transport.InboundHandler.doHandleResponse(InboundHandler.java:423)
                at app//org.elasticsearch.transport.InboundHandler$2.doRun(InboundHandler.java:391)
                at app//org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1044)
                at app//org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
                at java.base@24-ea/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)
                at java.base@24-ea/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)
                at java.base@24-ea/java.lang.Thread.run(Thread.java:1447)

                Caused by:
                java.lang.IllegalArgumentException: remote node [{TS_B}{TS_B}{w9X-moREQfG9mwpHw9c1Bw}{TS_B}{127.0.0.1}{127.0.0.1:9301}{9.1.0}{8000099-9010000}] is build [unknown-modified] of version [9.1.0] but this node is build [unknown] of version [9.1.0] which has an incompatible wire format
                    at org.elasticsearch.transport.TransportService$HandshakeResponse.throwOnIncompatibleBuild(TransportService.java:711)
                    at org.elasticsearch.transport.TransportService$HandshakeResponse.maybeThrowOnIncompatibleBuild(TransportService.java:696)
                    at org.elasticsearch.transport.TransportService$HandshakeResponse.<init>(TransportService.java:690)
                    at org.elasticsearch.action.ActionListenerResponseHandler.read(ActionListenerResponseHandler.java:64)
                    at org.elasticsearch.action.ActionListenerResponseHandler.read(ActionListenerResponseHandler.java:26)
                    at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1490)
                    at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1477)
                    at org.elasticsearch.transport.InboundHandler.doHandleResponse(InboundHandler.java:420)
                    ... 6 more
```

--- a/server/src/main/java/org/elasticsearch/transport/TransportService.java
+++ b/server/src/main/java/org/elasticsearch/transport/TransportService.java
@@ -83,7 +83,7 @@ public class TransportService extends AbstractLifecycleComponent
     /**
      * A feature flag enabling transport upgrades for serverless.
      */
-    private static final String SERVERLESS_TRANSPORT_SYSTEM_PROPERTY = "es.serverless_transport";
+    static final String SERVERLESS_TRANSPORT_SYSTEM_PROPERTY = "es.serverless_transport";
     private static final boolean SERVERLESS_TRANSPORT_FEATURE_FLAG = Booleans.parseBoolean(
         System.getProperty(SERVERLESS_TRANSPORT_SYSTEM_PROPERTY),
         false

--- a/server/src/test/java/org/elasticsearch/transport/TransportServiceHandshakeTests.java
+++ b/server/src/test/java/org/elasticsearch/transport/TransportServiceHandshakeTests.java
@@ -377,7 +377,7 @@ public class TransportServiceHandshakeTests extends ESTestCase {
     public void testAcceptsMismatchedServerlessBuildHash() {
         assumeTrue("Current build needs to be a snapshot", Build.current().isSnapshot());
         assumeTrue("Security manager needs to be disabled", System.getSecurityManager() == null);
-        System.setProperty("es.serverless", Boolean.TRUE.toString());   // security manager blocks this
+        System.setProperty(TransportService.SERVERLESS_TRANSPORT_SYSTEM_PROPERTY, Boolean.TRUE.toString());   // security manager blocks this
         try {
             final DisruptingTransportInterceptor transportInterceptorA = new DisruptingTransportInterceptor();
             final DisruptingTransportInterceptor transportInterceptorB = new DisruptingTransportInterceptor();
@@ -404,7 +404,7 @@ public class TransportServiceHandshakeTests extends ESTestCase {
             AbstractSimpleTransportTestCase.connectToNode(transportServiceA, transportServiceB.getLocalNode(), TestProfiles.LIGHT_PROFILE);
             assertTrue(transportServiceA.nodeConnected(transportServiceB.getLocalNode()));
         } finally {
-            System.clearProperty("es.serverless");
+            System.clearProperty(TransportService.SERVERLESS_TRANSPORT_SYSTEM_PROPERTY);
         }
     }
 

--- a/server/src/test/java/org/elasticsearch/transport/TransportServiceHandshakeTests.java
+++ b/server/src/test/java/org/elasticsearch/transport/TransportServiceHandshakeTests.java
@@ -377,7 +377,8 @@ public class TransportServiceHandshakeTests extends ESTestCase {
     public void testAcceptsMismatchedServerlessBuildHash() {
         assumeTrue("Current build needs to be a snapshot", Build.current().isSnapshot());
         assumeTrue("Security manager needs to be disabled", System.getSecurityManager() == null);
-        System.setProperty(TransportService.SERVERLESS_TRANSPORT_SYSTEM_PROPERTY, Boolean.TRUE.toString());   // security manager blocks this
+        System.setProperty(TransportService.SERVERLESS_TRANSPORT_SYSTEM_PROPERTY, Boolean.TRUE.toString());   // security manager blocks
+                                                                                                              // this
         try {
             final DisruptingTransportInterceptor transportInterceptorA = new DisruptingTransportInterceptor();
             final DisruptingTransportInterceptor transportInterceptorB = new DisruptingTransportInterceptor();

